src/luarocks/fetch.lua

```diff
@@ -235,6 +235,10 @@ function fetch.load_local_rockspec(filename, quick)
    if rockspec.source.cvs_module then rockspec.source.module = rockspec.source.cvs_module end
    if rockspec.source.cvs_tag then rockspec.source.tag = rockspec.source.cvs_tag end
 
+   if rockspec.source.revision and not deps.format_is_at_least("3.0") then
```

Comment:
With the version checking for the `revision` field in the type checker, you don't need to version-check again here.